src/dda/tools/git.py

```diff
@@ -66,3 +73,152 @@ def author_email(self) -> str:
             return env_email
 
         return self.capture(["config", "--get", "user.email"]).strip()
+
+    # === PRETEMPLATED COMMANDS === #
+    def get_remote_details(self, remote_name: str = "origin") -> Remote:
+        """
+        Get the details of the given remote for the Git repository in the current working directory.
+        The returned tuple is (org, repo, url).
+        """
+
+        remote_url = self.capture(
+            ["config", "--get", f"remote.{remote_name}.url"],
+        ).strip()
+
+        return Remote.from_url(remote_url)
+
+    def get_head_commit(self) -> Commit:
+        """
+        Get the current HEAD commit of the Git repository in the current working directory.
+        """
+        from dda.utils.git.commit import Commit
+
+        sha1 = self.capture(["rev-parse", "HEAD"]).strip()
+
+        return Commit(sha1=sha1)
+
+    def get_commit_details(self, sha1: str) -> CommitDetails:
+        """
+        Get the details of the given commit in the Git repository in the current working directory.
+        """
+        from datetime import datetime
+
+        from dda.utils.git.commit import CommitDetails
+
+        raw_details = self.capture([
+            "show",
+            "--quiet",
+            # Use a format that is easy to parse
+            # fmt: author name, author email, author date, parent SHAs, commit message body
+            "--format=%an%n%ae%n%ad%n%P%n%B",
+            "--date=iso-strict",
+            sha1,
+        ])
+        author_name, author_email, date_str, parents_str, *message_lines = raw_details.splitlines()
+
+        return CommitDetails(
+            author_name=author_name,
+            author_email=author_email,
+            datetime=datetime.fromisoformat(date_str),
+            message="\n".join(message_lines).strip().strip('"'),
+            parent_shas=list(parents_str.split()),
+        )
+
+    def commit_file(self, path: Path, *, content: str, commit_message: str) -> None:
+        """
+        Create and commit a single file with the given content.
+        """
+        path.write_text(content)
+        self.run(["add", str(path)])  # Will fail if path is not under cwd
+        self.run(["commit", "-m", commit_message])
```

Comment:
Let's extract these out into individual methods `add` & `commit` that this method could call. The former would accept `paths: list[Path]`.